runtime/ratelimit.c

```diff
@@ -215,7 +215,7 @@ ratelimitMsg(ratelimit_t *ratelimit, smsg_t *pMsg, smsg_t **ppRepMsg)
 	/* Only the messages having severity level at or below the
 	 * treshold (the value is >=) are subject to ratelimiting. */
 	if(ratelimit->interval && (pMsg->iSeverity >= ratelimit->severity)) {
-		if(withinRatelimit(ratelimit, pMsg->ttGenTime) == 0) {
+        if(withinRatelimit(ratelimit, pMsg->ttGenTime, getAPPNAME(pMsg, 0)) == 0) {
```

Comment:
Does not the `ratelimit->name` contain enough information to distinguish that? Or do you think something more should be added in name construction in `ratelimitNew()`?